tests/test_model.py

```diff
@@ -337,3 +346,18 @@ def test_dynamic_model_get_fields():
     fields = model.get_fields()
     assert '_private' not in fields
     assert len(fields) == 3
+
+
+def test_build_many(many_source):
+    models = BaseModel.build_many(many_source)
+
+    assert len(models) == 3
+    assert models[0].foo == '1 foo'
+    assert models[1].bar == '2 bar'
+    assert models[2].qux == '3 qux'
+
+
+def test_build_many_empty_iterable():
+    models = BaseModel.build_many([])
```

Comment:
I haven't thought about this usecase... I beleive we should raise an exception when this happens.